packages/trader/src/AppV2/Components/TradeParameters/Duration/container.tsx

```diff
@@ -0,0 +1,126 @@
+import React, { useEffect, useState } from 'react';
+import { ActionSheet, CaptionText, Text } from '@deriv-com/quill-ui';
+import { Localize } from '@deriv/translations';
+import { useTraderStore } from 'Stores/useTraderStores';
+import DurationEndTimePicker from './datepicker';
+import { observer } from '@deriv/stores';
+import DurationChips from './chips';
+import DurationWheelPicker from './wheelpicker';
+
+const DurationActionSheetContainer = observer(
+    ({ selected_hour, setSelectedHour }: { selected_hour: number[]; setSelectedHour: (arg: number[]) => void }) => {
+        const { duration, duration_unit, duration_units_list, onChangeMultiple, expiry_time } = useTraderStore();
+        const [unit, setUnit] = useState(expiry_time ? 'et' : duration_unit);
+        const [selected_time, setSelectedTime] = useState([duration]);
+        const [expiry_date_data, setExpiryDate] = useState<Date>(new Date());
+        const [end_time, setEndTime] = useState<string>('');
+        const [toggle_picker, setTogglePicker] = useState<boolean>(false);
+
+        const onChangeUnit = (value: string) => {
+            if (unit !== 'h') {
+                setSelectedHour([]);
+            }
+            setUnit(value);
+        };
+
+        const handleSelectExpiryDate = (date: Date) => {
+            setExpiryDate(date);
+            const current_date = new Date();
+            const timeDifference = +date - +current_date;
+            const dayDifference = Math.ceil(timeDifference / (1000 * 60 * 60 * 24));
```

Comment:
nit: same as current_date above:)
```suggestion
            const time_difference = +date - +current_date;
            const day_difference = Math.ceil(timeDifference / (1000 * 60 * 60 * 24));
```